src/shape.cpp

```diff
@@ -151,6 +152,24 @@ struct shape_impl
             m_lens.begin(), m_lens.end(), std::size_t{1}, std::multiplies<std::size_t>());
     }
 
+    std::size_t get_index(size_t i) const
+    {
+        std::size_t result = 0;
+        std::size_t s      = 1;
+        std::size_t k      = m_lens.size();
+
+        while(k != 0)
```

Comment:
Actually would be cleaner to do `for(auto k:reverse(range(m_lens.size()))`

Reply:
In c++ 17.. ?

Reply:
Yes we have these functions under the `migraphx/ranges.hpp` header.

Reply:
Looking at the code, (ranges.hpp): reverse(range()) would create a new range objects (twice)..

Reply:
Of course, its two function calls.

Reply:
Done. 
(arguably.. a complex step for a simple original one..)